lib/floe/workflow/states/input_output_mixin.rb

```diff
@@ -15,9 +15,8 @@ def process_output(context, results)
           return if output_path.nil?
 
           results = result_selector.value(context, results) if @result_selector
-          if result_path.payload.start_with?("$.Credentials")
-            credentials = result_path.set(context.credentials, results)["Credentials"]
-            context.credentials.merge!(credentials)
+          if result_path.payload.start_with?("$$.Credentials")
```

Comment:
Very minor, but I think we need to be a bit more pedantic here.  Technically it mattered with the old code more, but if someone had something liek `$.CredentialsStore`, this would have previously triggered and shouldn't have.  Think this should be something like.

```suggestion
          if result_path.payload.match?(/^$$.Credentials\b/)
```

Reply:
So you can't normally assign anything to the Context object, we're making an exception here for `$$.Credentials` which means it isn't possible for someone to have `"ResultPath": "$$.CredentialsStore"` and have it be valid (this would fail previously but I've added a better validation that prints a helpful error message.)